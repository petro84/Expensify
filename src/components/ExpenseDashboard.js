import React from 'react';

import ExpenseSummary from './ExpenseSummary';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';

const ExpenseDashboard = () => (
  <div>
    <ExpenseSummary />
    <ExpenseListFilters />
    <ExpenseList />
  </div>
);

export default ExpenseDashboard;
