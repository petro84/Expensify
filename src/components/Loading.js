import React from 'react';
import loader from '../images/loader.gif';

const Loading = () => (
    <div className='loader'>
        <img className='loader__image' src={loader} />
    </div>
);

export default Loading;