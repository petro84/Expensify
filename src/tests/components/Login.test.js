import React from 'react';
import { shallow } from 'enzyme';

import { Login } from '../../components/Login';

let startLogin, wrapper;

beforeEach(() => {
    startLogin = jest.fn();
    wrapper = shallow(<Login startLogin={startLogin} />);
});

test('should render Login correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should call startLogin on click', () => {
    wrapper.find('button').simulate('click');
    expect(startLogin).toHaveBeenCalled();
})