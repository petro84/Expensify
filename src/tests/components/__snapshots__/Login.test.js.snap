// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Login correctly 1`] = `
<div>
  <button
    onClick={[MockFunction]}
  >
    Login
  </button>
</div>
`;
